pkg/driver/kubernetes/driver.go

```diff
@@ -262,6 +267,55 @@ func (d *Driver) RuntimeSockProxy(ctx context.Context, name string) (net.Conn, e
 
 }
 
+func (d *Driver) GetVersion(ctx context.Context) (string, error) {
+	restClient := d.clientset.CoreV1().RESTClient()
+	restClientConfig, err := d.KubeClientConfig.ClientConfig()
+	if err != nil {
+		return "", err
+	}
+	pod, _, err := d.podChooser.ChoosePod(ctx)
+	if err != nil {
+		return "", err
+	}
+	if len(pod.Spec.Containers) == 0 {
+		return "", errors.Errorf("pod %s does not have any container", pod.Name)
+	}
+	containerName := pod.Spec.Containers[0].Name
+	cmd := []string{"buildkitd", "--version"}
+	buf := &bytes.Buffer{}
+
+	req := restClient.
+		Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: containerName,
+			Command:   cmd,
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec)
+	u := req.URL()
+	exec, err := remotecommand.NewSPDYExecutor(restClientConfig, "POST", u)
+	if err != nil {
+		return "", err
+	}
+	// TODO how to timeout if something goes bad...?
```

Comment:
does remotecommand not support using a context?

Reply:
Thanks for asking the question - digging a bit deeper, yes, it looks like it's possible on the Request object.  I'll update it.